src/ORM/Table.php

```diff
@@ -2086,6 +2086,61 @@ public function saveMany(iterable $entities, $options = [])
         return $entities;
     }
 
+    /**
+     * Persists multiple entities of a table.
+     *
+     * The records will be saved in a transaction which will be rolled back if
+     * any one of the records fails to save due to failed validation or database
+     * error.
+     *
+     * @param \Cake\Datasource\EntityInterface[]|\Cake\Datasource\ResultSetInterface $entities Entities to save.
+     * @param array|\ArrayAccess $options Options used when calling Table::save() for each entity.
+     * @return \Cake\Datasource\EntityInterface[]|\Cake\Datasource\ResultSetInterface Entities list.
+     * @throws \Exception
+     * @throws \Cake\ORM\Exception\PersistenceFailedException
+     */
+    public function saveManyOrFail(iterable $entities, $options = [])
```

Comment:
There's lot of code duplication between this method and `saveMany()`, only difference being this method stores reference to entity which failed to save and then throwing exception. So the common code should be consolidated into a helper method.

Reply:
I refactored.
We shouldnt use too many exceptions for workflow :) But I hope the current solution is maximum DRY while still allowing both cases.